### PR TITLE
PE1670, filters added, tests added

### PR DESCRIPTION
**Pull request by DanielKerri-WDFC**
review/feedback before merge

### CODE REVIEW

File: HealthMonitoring.SelfHost/Content/Home/home.html [hunk 1]
```diff
@@ -42,36 +42,71 @@
...
-                <span ng-repeat="status in filters[statusFilterName]" ng-click="removeItemFromFilter(status, statusFilterName)" class="endpoint-status status-{{status}} endpoint-clickable">{{status}}</span>
+            
+            
+        </article>
+        <article>
+            <header>Filters</header>
+            <p ng-if="endpoints==null">-- connection issue --</p>
+            <p ng-if="endpoints.length==0">No endpoints registered yet</p>
+            
+            <div class="">
```
Comment (Pawlyha) [inline HealthMonitoring.SelfHost/Content/Home/home.html]: I think class attr can be removed

File: HealthMonitoring.SelfHost/HealthMonitoring.SelfHost.csproj [hunk 2]
```diff
@@ -15,6 +15,21 @@
     <SolutionDir Condition="$(SolutionDir) == '' Or $(SolutionDir) == '*Undefined*'">..\</SolutionDir>
     <NuGetPackageImportStamp>
     </NuGetPackageImportStamp>
+    <PublishUrl>publish\</PublishUrl>
```
Comment (andronz) [inline HealthMonitoring.SelfHost/HealthMonitoring.SelfHost.csproj]: what is this ?

File: HealthMonitoring.SelfHost/HealthMonitoring.SelfHost.csproj [hunk 3]
```diff
@@ -239,7 +254,18 @@
       <Name>HealthMonitoring</Name>
     </ProjectReference>
   </ItemGroup>
-  <ItemGroup />
+  <ItemGroup>
```
Comment (Pawlyha) [inline HealthMonitoring.SelfHost/HealthMonitoring.SelfHost.csproj]: What is this?

Comment (odekhtiarenko) [inline HealthMonitoring.SelfHost/HealthMonitoring.SelfHost.csproj]: What is this?

File: HealthMonitoring.AcceptanceTests/Scenarios/Selenium/Home_page.Steps.cs [hunk 4]
```diff
@@ -186,6 +188,48 @@ public void Then_only_endpoints_with_chosen_parameters_should_be_shown()
...
+            IList<IWebElement> groupElements = _driver
+                .WaitElementsAreRendered(
+                    By.XPath(_endpointsGroupsSelectorValue), 
+                    group => group.Displayed && group.Enabled)
+                .ToList();
+
+            bool areStatusesCorrect = groupElements.All(m => m.Text.ToLower().IndexOf(groupFilterValue, StringComparison.OrdinalIgnoreCase) >= 0);
+
+            Assert.True(areStatusesCorrect);
+            Assert.True(groupElements.Count(m => m.Text.ToLower().IndexOf(groupFilterValue, StringComparison.OrdinalIgnoreCase) >= 0) == groupElements.Count);
```
Comment (Pawlyha) [inline HealthMonitoring.AcceptanceTests/Scenarios/Selenium/Home_page.Steps.cs]: m.Text.ToLower().IndexOf(...) => m.Text.Contains ?

File: HealthMonitoring.SelfHost/Content/Home/home.html (hunk 1 above)
Comment (DanielKerri-WDFC) [inline HealthMonitoring.SelfHost/Content/Home/home.html]: Will remove

File: HealthMonitoring.AcceptanceTests/Scenarios/Selenium/Home_page.Steps.cs (hunk 4 above)
Comment (DanielKerri-WDFC) [inline HealthMonitoring.AcceptanceTests/Scenarios/Selenium/Home_page.Steps.cs]: Contains is case sensitive

Comment (Pawlyha) [inline HealthMonitoring.AcceptanceTests/Scenarios/Selenium/Home_page.Steps.cs]: there is extension method (in Linq): m.Text.Contains("sdf", StringComparer.OrdinalIgnoreCase))
that takes comparer as 2 parameter

File: HealthMonitoring.SelfHost/HealthMonitoring.SelfHost.csproj (hunk 3 above)
Comment (DanielKerri-WDFC) [inline HealthMonitoring.SelfHost/HealthMonitoring.SelfHost.csproj]: Correction

Comment (DanielKerri-WDFC) [inline HealthMonitoring.SelfHost/HealthMonitoring.SelfHost.csproj]: Correction

**Pull request merged**